contracts/src/Summa.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.18;
+
+// Uncomment this line to use console.log
+//import "hardhat/console.sol";
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface Verifier {
+    function verify(
+        uint256[] calldata pubInputs,
+        bytes calldata proof
+    ) external view returns (bool);
+}
+
+contract Summa is Ownable {
+    using ECDSA for bytes32;
+
+    uint256 constant ETH_INPUTS = 1;
+    uint256 constant ERC20_INPUTS = 1;
+
+    struct PublicInput {
```

Comment:
Can we separate the actual public inputs of the circuits from the variable that participate to the proof of solvency but are not strictly public inputs of the circuit? Such as the addresses and the signatures